cmake/macos/compilerconfig.cmake

```diff
@@ -15,6 +15,21 @@ if(NOT CMAKE_OSX_ARCHITECTURES)
 endif()
 set_property(CACHE CMAKE_OSX_ARCHITECTURES PROPERTY STRINGS arm64 x86_64)
 
+if(NOT OBS_MACOS_MINIMUM_SDK OR NOT OBS_MACOS_MINIMUM_XCODE)
+  message(FATAL_ERROR "Required variable not found.")
+endif()
```

Comment:
Wondering if this is fine or if we should explicify the preset situation somewhere else anyways.
Problem is that otherwise, an empty string is always a lower version than a string with a version, so we wouldn't catch the people who don't set presets.

Reply:
As discussed off-thread I'd actually lean towards hardcoding those values in CMake code and not use cache variables from the preset.

The expectation for cache variables is that they are set by the developer (the preset is just a convenience feature), so implementing this kinda expects the developer to always provide a minimum SDK and Xcode version.

But we don't want to make this a developer choice, it is a hard limit set by the project. As such it should be coded directly into the `compilerconfig` file.

Reply:
This should be addressed.